CONTRIBUTING.md

```diff
@@ -93,6 +94,38 @@ Refer to the
 [Semantic Convention YAML Language](https://github.com/open-telemetry/build-tools/blob/v0.25.0/semantic-conventions/syntax.md)
 to learn how to make changes to the YAML files.
 
+#### Code structure
+
+The YAML (model definition) and Markdown (documentation) files are organized in the following way:
+
+```
+├── docs
+│   ├── attribute_registry
+│   ├── {root-namespace}
+│   │   ├── README.md
+│   │   ├── ....md
+├── model
+│   ├── {root-namespace}
+│   │   ├── {root-namespace}-events.yaml
```

Comment:
Do we need to force `{root-namespace}` in all the file names?  Should we instead make sure directory is easy to see in provenance annotations/error messages?

Reply:
I'd like to enforce it and have the root namespace for two reasons:

- navigation - the way I navigate in the repo is go-to-file in vscode and on GH based on the name and having ~100 files called `spans` or `registry` would be hard.
- consistency (minor-ish)

I don't think it's super-critical, but also having it does not seem to hurt.

PS: if someone has a better way to navigate, please show me, the struggle is real

Reply:
> navigation - the way I navigate in the repo is go-to-file in vscode and on GH based on the file name and having ~100 files called spans.yaml or registry.yaml would be hard.

fwiw, in Intellij whatever you type in go-to-file is applied across both directories and files, so typing `databasemssql` will get you to `docs/database/mssql.md`

Reply:
wow, thanks @trask, it's a game-changer!